### DISCUSSION TITLE
Qual a rotina de estudos de vocês? Como fazem? O que deu mais certo?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1566] [Discussão]
Galera,

estou com um certo dilema, tentando montar uma rotina na qual eu consiga me desenvolver em habilidade que não uso hoje no atual trabalho...

Atualmente trabalho como  fullstack .Net (senior) / ReactJS (junior), já alcancei certa "estabilidade" na tecnologia .net, e o que preciso fazer para atender a empresa quanto a código é tranquilo, em react ainda aprendendo muitas coisas mas já colocando projetos em produção tranquilo.

Mas quero evoluir em outros pontos que não usarei na empresa, quero alcançar a stack: .Net Core + NodeJS + ReactJS + React-Native, e estes dois últimos terei de estudar e praticar fora de meu expediente.

Estou pensando em acordar por volta das 5h, e estudar todos os dias úteis até às 8h, horário em que saio para ir ao trabalho na empresa. E aos finais de semana seprar tempo para praticar criando projetos pessoais em react-native e nodejs.

O que acham? Como indicariam?  Como estudam?
E acham que faz sentido esta stack?

(Já trabalhei com php por 2 anos, mas não incluo ele na stack por focar em .Net e NodeJS atualmente)

Abs.

**Comment by Cleysonlb** [comment]
Falae @wilsonneto-dev beleza?
Poxa cara bem legal sua preocupação em aprender coisas novas.
Primeiro seria interessante vc entender qual modelo melhor se adapta pra você na questão de conteudo, vc se da melhor com video aulas, livros, posts, tutoriais?
Identificando isso é legal vc montar um roteirinho, nada muito grandioso, somente para você ter foco no que estudar e como avançar, exemplo:

### React
- Javascript
- JSX
- Padrões
- Teste

Outra coisa que ajuda muito é por a mão na massa e aplicar oq vc aprender, fazendo pequenos projetinhos, digo pequeno pq eu mesmo ja cai na ideia de criar coisas muito complexas e esquecia que o foco é a aplicação em si do que eu aprendi e não muito oq estou construindo hahah.

Sobre seus horarios, acho que de cara demandar 3 horas (de 5h às 8h) um pouco puxado, ainda mais pensando que você hoje não possui esse hábito, tenta começar com 1 hora por dia, mais importante que a quantidade de horas que estuda de uma vez, é a frequência que estuda dia a dia!

Espero ter ajudado, e boa sorte com seus estudos :)

**Comment by ninetails** [comment]
Puxado depende do referencial. Se vc ainda não tiver família, seria como trocar o horário que lá no começo vc gastava com facul pelos estudos.

Creio que o que mais consideraria desafiador seria se manter motivado. Infelizmente é algo que vc vai ter que criar por si só, pq não vai ter um diploma lá na frente para servir de cenoura e pelo que entendi, não é nada que vc queira pra usar no trampo atual e está satisfeito em continuar nele. Outro problema seria como organizar seus estudos de forma eficiente.

Dado estes dois problemas que ao meu ver seriam os mais críticos, creio que o ideal é vc setar alguma meta paupável de um jeito que te cobre mas que também não desanime. Tal como construir um produto, fazer _milestones_ pros teus estudos, e no final uma recompensa pra talvez tentar se motivar.

Pessoalmente, diria que todo estudo vale a pena mas também levando em conta preferências pessoais de cada um. Se não é necessariamente ligado ao trampo ou a um novo trampo, diria até para não se importar se faz sentido, como um _hobby_.

**Comment by jpferreiradev** [comment]
Eu também estou passando por esse dilema, mas precisamente criar uma rotina 
de estudos, atualmente estou trabalhando em um estágio que é full-stack, mas 
estamos começando pelo front, que é feito em Angular. A stack é Java/Spring + Angular.
Eu ainda não sei realmente aonde eu aprendo mais se é livros ou videos, o mais xato
é achar organizar os horários, e ainda tem que estudar para a faculdade que é em outra 
linguagem, como vocês faziam para organizar os horários de estudo? Porque infezlimente estudar 
para a faculdade desanima demais, eu só quero focar nos assuntos do trabalho, kkkkk

**Comment by wilsonneto-swe** [comment]
Valeu galera pelas opiniões, para organizar meu fluxo de estudos estou seguindo dois cursos e a própria documentação do Node e do React-Native, de inicio focado em React Native, logo após irei entrar em Node.

Sempre passando a documentação -> um curso -> prototipagens -> lançar um mvp no mercado (mesmo que algo gratuito como o app que farei para doação/adoção de cães e gatos...)

E imagino que já estando a um tempo com linguagens de back (PHP e .Net) e também no front com ReactJS, HTML e CSS, acredito e sinto que é bem mais fácil aprender novas tecnologias, e inclusive acho que começar a brincar com NodeJS me ajudou a melhorar no .net e vice-versa, além de que DDD, CQRS e vários conceitos de arquitetura que hoje considero até mais importantes que a linguagem em sim da para usar em qualquer linguagem...

Vou tentar fazer esta rotina de 3h/dia e fazer uma "ementa" mensal de acordo com meus objetivos.

Valeu @ninetails, @Cleysonlb e @jotepauder pelas opiniões.

Abs

**Comment by eliseumds** [comment]
Como o @jotepauder falou: rotina.

Eu, ultimamente, tenho feito uns cursos da Udemy (um por vez), pratico com side projects (invista uns dólares na Amazon pra se familiarizar, vale a pena) e escrevendo notas sobre o que eu aprendi.

* Coloca um site no ar com domínio próprio
* Configura HTTPS, de preferência com renovação automática
* Configura deploy automático
* Cria pipelines de testes
* Separa ambientes de dev e prod
* Sirva conteúdo via CDN (com domínio próprio ex `cdn.meusite.com.br`) e caching
* Cria um outro site em outro domínio e faz integração entre os dois (via scripts e webhooks). Isso pode te ensinar muito sobre CORS, OAuth e Content-Security-Policy

É impressionante a quantidade de coisas que aprendemos quando botamos a mão na massa, coisas que não são ensinadas em cursos. Boa sorte!

**Comment by bbertolani** [comment]
@eliseumds cara essa parte de por a mão na massa é fundamental, gostei da ideia de gastar uns dolares na Amazon, meio por cima quanto você gastou nessa brincadeira?

**Comment by eliseumds** [comment]
@bbertolani vai depender do que você estiver usando.

Um domínio comprado na Amazon (AWS Route53) vai custar uns 10-15 dólares por ano.

A maioria dos serviços básicos são cobertos pelo Free Tier por um ano (AWS, GCloud e Azure oferecem). Muitos serviços populares estão incluídos. Kubernetes, por ex, é uma exceção, então tenha o cuidado de destruir o cluster após terminar a brincadeira. Não deve sair mais de 15-20 dólares pra brincar um pouco com Kubernetes.

Você pode criar alertas de custos pra ter mais controle: https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/tracking-free-tier-usage.html#free-budget.

Houve uma vez onde eu fiz cagada e criei uma Lambda function que rodava recursivamente. Ela rodava após um objeto ser adicionado a um bucket S3, gerava uma nova imagem e a adicionava a esse mesmo bucket, boom, chamando a função de novo, infinitamente. Perdi uns 30 dólares em alguns minutos :S

**Comment by lucas-moraes** [comment]
Cara, eu estudo programação praticamente todos os dias, estou em transição de carreira, de contador para Desenvolvedor. **Então trabalho das 8 às 18, depois das 19:30 às 00:00 estudo**, o que me complica um pouco é estudar e fazer Challengers.

O que eu faço não é recomendável, por que corre-se o risco de me tornar um programador "meia boca", eu gosto de ver as coisas funcionando, tipo construir e testar, tentativa e erro, assim vou construindo "tijolinho por tijolinho" o conhecimento, por exemplo fiz um projeto Fullstack em HTML, CSS, JQuery, PHP, SQL, [Cloudcont.app](https://cloudcont.app). 
Agora meu objetivo é atualizar esse projeto para JAMStack.

_Já li que o certo é primeiro ler todo a Documentação da linguagem e depois desenvolver algo, mas eu prefiro fazer da forma mais legal para mim._
